cdap-ui/app/cdap/components/AbstractWidget/Comment/CommentBox.tsx

```diff
@@ -0,0 +1,164 @@
+  return {
+    root: {
+      zIndex: 1,
+      cursor: 'text',
+      userSelect: 'text',
+    },
+    contentRoot: {
+      paddingTop: ({ editMode }) => (!editMode ? 0 : theme.spacing(2)),
+    },
+    cardAction: {
+      display: 'inline-flex',
+    },
+    textarea: {
+      width: '100%',
+      resize: 'vertical',
+      maxHeight: COMMENT_WIDTH, // Allow to resize vertically upto 300px
+      padding: theme.spacing(2),
+    },
+  };
+});
+interface ICommentStyleProps {
+  editMode: boolean;
+}
+
+interface ICommentBoxProps {
+  comment?: IComment;
+  onSave?: (comment?: IComment) => void;
+  onDelete?: () => void;
+  focus?: boolean;
+  disabled?: boolean;
+}
+
+export default function CommentBox({
+  comment,
+  onSave,
+  onDelete,
+  focus = false,
+  disabled = false,
+}: ICommentBoxProps) {
+  const isThereExistingComment = comment.content && comment.content.length > 0;
+  const [editMode, setEditMode] = React.useState(!isThereExistingComment);
+  const [localComment, setLocalComment] = React.useState(comment);
+
+  const classes = useStyles({ editMode });
+
+  const onCommentChange = (value: string) => {
+    if (!localComment.content || !localComment.content.length) {
+      setLocalComment({
+        ...comment,
+        content: value,
+        createDate: Date.now(),
+      });
+    } else {
+      setLocalComment({
+        ...comment,
+        content: value,
+      });
+    }
+  };
+  const onSaveComment = () => {
+    onSave({
+      ...comment,
+      content: localComment.content,
+    });
+  };
+  const onResetComment = () => {
+    onSave(comment);
+  };
+
+  React.useEffect(() => {
+    setLocalComment(comment);
+    if (!comment.content.length) {
+      setEditMode(true);
+    }
+  }, [comment]);
+
+  const CommentMenuLocal = disabled ? null : (
+    <CommentMenu onEdit={() => setEditMode(true)} onDelete={onDelete} />
+  );
+  if (!editMode) {
+    return (
+      <Card className={classes.root}>
+        <CardHeader
+          avatar={<AccountCircleOutlinedIcon />}
+          action={CommentMenuLocal}
+          title={'Comment'}
+          subheader={humanReadableDate(comment.createDate, true) || '--'}
+        />
+        <CardContent className={classes.contentRoot}>
+          <Markdown markdown={`${localComment.content}`} />
+        </CardContent>
+      </Card>
+    );
+  }
+  return (
+    <Card className={classes.root}>
+      <CardContent className={classes.contentRoot}>
+        <TextareaAutosize
+          className={classes.textarea}
+          value={localComment.content}
+          placeholder="Add a comment"
+          rowsMax={10}
+          rowsMin={3}
+          onChange={(e) => onCommentChange(e.target.value)}
+          autoFocus={focus}
+        />
+      </CardContent>
+      <CardActions className={classes.cardAction}>
+        <Button
+          variant="contained"
+          color="primary"
+          disabled={localComment.content.length === 0}
```

Comment:
the props for comment indicates that `comment` can be nullable - this may throw error

Reply:
This doesn't happen here as the `CommentBox` is used from within the `DynamicAnchoredComments` component which always sets a local comment before passing on to CommentBox. But I have added extra check in case this component is used elsewhere.